### PR TITLE
Make sure relevant std::stringstreams are set to binary

### PR DESCRIPTION
**Pull request by sfan5**
Honestly if the binary flag actually did much I bet more things would be broken sooner.

Also fixed some unnecessary copies in places, for example in every(?) map backend.

## To do

This PR is a Ready for Review.

### CODE REVIEW

**Review by nerzhul — APPROVED** [review]

**Comment by nerzhul** [comment]
That should improve a very very bit much perf for leveldb and pg on load

File: src/script/common/c_converter.cpp [hunk 1]
```diff
@@ -76,10 +76,7 @@ static void set_vector_metatable(lua_State *L)
 
 void push_float_string(lua_State *L, float value)
 {
-	std::stringstream ss;
-	std::string str;
-	ss << value;
-	str = ss.str();
+	auto str = ftos(value);
 	lua_pushstring(L, str.c_str());
```
Comment (Desour) [inline src/script/common/c_converter.cpp:80]:
```suggestion
	lua_pushlstring(L, str.data(), str.size());
```

File: src/script/lua_api/l_util.cpp [hunk 2]
```diff
@@ -272,11 +272,11 @@ int ModApiUtil::l_compress(lua_State *L)
...
 	int level = -1;
-	if (!lua_isnone(L, 3) && !lua_isnil(L, 3))
-		level = readParam<float>(L, 3);
+	if (!lua_isnoneornil(L, 3))
+		level = readParam<int>(L, 3);
 
-	std::ostringstream os;
-	compressZlib(std::string(data, size), os, level);
+	std::ostringstream os(std::ios_base::binary);
+	compressZlib(reinterpret_cast<const u8*>(data), size, os, level);
```
Comment (Desour) [inline src/script/lua_api/l_util.cpp:279]:
```suggestion
	compressZlib(reinterpret_cast<const u8 *>(data), size, os, level);
```
^ Pedanticism.

(Btw., I find it awful how the 2nd argument is just ignored here, to silently ignore mod errors. But that's out of scope.)

**Review by Desour — COMMENTED** [review]
:+1:

File: src/script/common/c_converter.cpp (hunk 1 above)
Comment (SmallJoker) [reply src/script/common/c_converter.cpp:80]: Why? It's a float. It cannot contain NULL characters.

Comment (sfan5) [reply src/script/common/c_converter.cpp:80]: It would still save Lua from doing a `strlen`, pretty irrelevant in terms of performance though.

Comment (Desour) [reply src/script/common/c_converter.cpp:80]: But it's probably a tiny little bit faster if it gets the size.
(Not that it really matters, this function is just called from `push_noiseparams`.)

**Pull request merged**